lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -451,6 +452,7 @@ public void onMessage(ITransport transport, ProtocolMessage message) throws Ably
 		if (Log.level <= Log.VERBOSE)
 			Log.v(TAG, "onMessage(): " + message.action + ": " + new String(ProtocolSerializer.writeJSON(message)));
 		try {
+			lastActivity = System.currentTimeMillis();
```

Comment:
If we don't allow for WebSocket heartbeats we're now going to incorrectly move people into the suspended state now.

Reply:
In fact there already is support for tracking activity at the transport level, which includes heartbeats:
https://github.com/ably/ably-java/blob/develop/lib/src/main/java/io/ably/lib/transport/WebSocketTransport.java#L188

The staleness check needs to be based on this instead of introducing a new mechanism here.